doc/design/OrionSTHSink.md

```diff
@@ -199,14 +215,57 @@ A configuration example could be:
 
-## Contact
+[Top](#top)
+
+##<a name="section4"></a>Implementation details
+###<a name="section4.1"></a>`OrionSTHSink` class
+`OrionSTHSink` extends `OrionMongoBaseSink`, which as any other NGSI-like sink extends the base `OrionSink`. The methods that are extended are by `OrionMongoBaseSink` are:
+
+    public void start();
+
+`MongoBackend` is created. This must be done at the `start()` method and not in the constructor since the invoking sequence is `OrionSTHSink()` (contructor), `configure()` and `start()`.
+
+    public void configure(Context);
+    
+A complete configuration as the described above is read from the given `Context` instance.
+
+The methods that are extended by `OrionSTHSink` are:
+
+    void persist(Map<String, String>, NotifyContextRequest) throws Exception;
+    
+The context data, already parsed by `OrionSink` in `NotifyContextRequest`, is iterated and persisted in the MongoDB backend by means of a `MongoBackend` instance. Header information from the `Map<String, String>` is used to complete the persitence process, such as the timestamp or the destination.
+
+[Top](#top)
+
+###<a name="section4.2"></a>`MongoBackend` class
+This is a convenience backend class for MongoDB that provides methods to persist the context data both in raw of aggregated format. Relevant methods regarding raw format are:
+
+    public void createDatabase(String dbName) throws Exception;
+    
+Creates a database, given its name, if not exists.
+    
+    public void createCollection(String dbName, String collectionName) throws Exception;
+    
+Creates a collection, given its name, if not exists in the given database.
+    
+    public void insertContextDataRaw(String dbName, String collectionName, long recvTimeTs, String recvTime, String entityId, String entityType, String attrName, String attrType, String attrValue, String attrMd) throws Exception;
+    
+Updates or inserts (depending if the document already exists or not) a set of documents in the given collection within the given database. Such a set of documents contains all the information regarding current and past notifications (historic) for a single attribute. a set of documents is managed since historical data is stored using several resolutions and range combinations (second-minute, minute-hour, hour-day, day-month and month-year). See STH at [Github](https://github.com/telefonicaid/IoT-STH/blob/develop/README.md) for more details.
+
+[Top](#top)
+
+##<a name="section5"></a>Contact
 Francisco Romero Bueno (francisco.romerobueno@telefonica.com) **[Main contributor]**
 <br>
 Fermín Galán Márquez (fermin.galanmarquez@telefonica.com) **[Contributor and Orion Context Broker owner]**
 <br>
 Germán Toro del Valle (german.torodelvalle@telefonica.com) **[Contributor]**
 <br>
-Iván Arias León (ivan.ariasleon@telefonica.com) **[Quality Assurance]**
\ No newline at end of file
+Iván Arias León (ivan.ariasleon@telefonica.com) **[Quality Assurance]**
+
+[Top](#top)
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/commit/51d342c1565cf1b6efa79da68154b241c77cb4b7